### DISCUSSION TITLE
FPS should adapt to zoom

### DISCUSSION THREAD

**Discussion by jorivaa** [Stellarium/stellarium#3702] [Ideas]
We should not waste compute, right?

When Stellarium is zoomed out it still updates the screen tens of times per second. That causes my fan to be on and my Power Meter shows the computer uses more electricity. At that zoom setting even 1 frame per _minute_ is plenty.

Obviously when zoomed very deep in, even full speed FPS isn't enough.

SUGGESTION: throttle the refresh rate to match zoom level.

**Comment by gzotti** [comment]
This is your system. There are too many systems to automate everything for you. Just use the available settings to throttle your system to e.g. 2fps (idle) and 10 fps (max).

**Reply by jorivaa** [reply]
I'm sorry if I came across as arrogant. Not intended.

I did some testing, and to my surprise I found 15 fps good for even watching the ISS fly across the Moon at 1 degree FOV. You might want to find a case which needs more FPS than that to look smooth. If not, then may I suggest that the defaults in the config file are changed to 15 for both min and max fps. -- I understand that showing FPS is good for debugging and ensuring quality of plugins, and that's why the default is practically _unlimited in development builds_. For normal use, wouldn't full speed be _better as opt-in than opt-out_, especially when no regular user would even notice the difference?

Why am I still here? I live in Finland, and we are constantly guilt-tripped for eating, living, driving, and breathing "because Climate Change". Well, as a developer, I could stop eating. But what if I could do more? If I knew how many people use a software and how many hours per year they use it, then it would be easy to calculate how big a difference something as "trivial" as a config-file default could be.

But no biggie, I won't push this any more. My background is over 40 years of Unix, and the ethos "don't waste cycles" is thus in my blood.  Maybe I see the defaults get smaller one day, **but either way, I think Stellarium is the best**!

**Reply by gzotti** [reply]
A common lament is that Stellarium is so slow (apparently on the "average PC out there"), so defaulting to 15fps max does not sound good as it would clip the upper end of first-time experience in times of "nobody reads instructions, all must go as users expect". 

Early 20th century movies were about 18fps and indeed flickering. Usual movie or TV rates are 24/25/30fps and were regarded sufficient for a century or so. I don't know why some folks need 200fps 4k gaming. (Are they insects?) If your system makes 60fps or more it is "modern" (likely 2020s with some dedicated hardware). Others have complained that Stellarium maxes out their telescope PC system from 2010 with 12fps. They were rejoycing that we finally added to the GUI the fps intent settings that had been available in the config.ini forever. 

So now, it's in the GUI and really a matter of taste. Maybe use a RaspberryPi4/5 and 720p screen to conserve energy, and not something with an 1000W PSU :-) Although the latter can help heating your house in winter...

**Reply by 10110111** [reply]
> Maybe I see the defaults get smaller one day

This is unlikely, given what Georg said above. But, given that Stellarium is free and open-source, you can play with the logic of changing frame rate inside `src/StelMainView.cpp`. Look for `minfps` and `maxfps` variables and how they are used. FoV can be found via `StelCore::getCurrentStelProjectorParams()`, a call to which in the same file can give you an example.

> I live in Finland, and we are constantly guilt-tripped for eating, living, driving, and breathing "because Climate Change".

You don't have to fall into this trap. Finland's emission of greenhouse gases is tiny compared to the most "efficient" climate changer, China. Even if Finland freezes out completely, this will have little effect on the climate. The whole EU emits just a bit smaller amount of GG than India (which is far behind China), while India is expected to increase its rate of air pollution with time ([source](https://edition.cnn.com/interactive/2023/12/us/countries-climate-change-emissions-cop28/)).
